internal/notifications/noop.go

```diff
@@ -0,0 +1,26 @@
+package notifications
+
+import (
+	"context"
+
+	"github.com/rs/zerolog"
+)
+
+type noopNotificationClient struct{}
+
+// var _ NotificationClient = &noopNotificationClient{}
```

Comment:
You can keep this, the reason is that the compiler will show you more readable error when you miss a method in the implementation. Or you can delete if you don’t like it.